orchestrator/src/main/java/com/yahoo/vespa/orchestrator/controller/ClusterControllerClientTimeouts.java

```diff
@@ -36,40 +36,28 @@ public class ClusterControllerClientTimeouts implements JaxRsTimeouts {
      * A logical call to CC may in fact call the CC several times, if the first ones are down and/or not
      * the master.
      *
-     * @param timeBudget  The time budget for a single logical call to the the Cluster Controller.
+     * @param timeBudget  The time budget for a single logical call to the Cluster Controller.
      */
     public ClusterControllerClientTimeouts(TimeBudget timeBudget) {
         this.timeBudget = timeBudget;
     }
 
-    @Override
-    public Duration getConnectTimeoutOrThrow() {
-        return CONNECT_TIMEOUT;
-    }
-
-    @Override
-    public Duration getReadTimeoutOrThrow() {
-        Duration timeLeft = timeBudget.timeLeft().get();
-
-        // timeLeft = CONNECT_TIMEOUT + readTimeout
-        Duration readTimeout = timeLeft.minus(CONNECT_TIMEOUT);
-
-        if (readTimeout.toMillis() <= 0) {
+    public Duration getServerTimeoutOrThrow() {
+        // readTimeout = DOWNSTREAM_OVERHEAD + serverTimeout
+        TimeBudget serverBudget = readBudget().withReserved(DOWNSTREAM_OVERHEAD);
+        if (serverBudget.timeLeft().get().compareTo(MIN_SERVER_TIMEOUT) < 0)
             throw new UncheckedTimeoutException("Timed out after " + timeBudget.originalTimeout().get());
-        }
 
-        return readTimeout;
+        return serverBudget.timeLeft().get();
     }
 
-    public Duration getServerTimeoutOrThrow() {
-        // readTimeout = DOWNSTREAM_OVERHEAD + serverTimeout
-        Duration serverTimeout = getReadTimeoutOrThrow().minus(DOWNSTREAM_OVERHEAD);
-
-        if (serverTimeout.toMillis() < MIN_SERVER_TIMEOUT.toMillis()) {
-            throw new UncheckedTimeoutException("Timed out after " + timeBudget.originalTimeout().get());
-        }
+    public Duration connectTimeout() {
+        return CONNECT_TIMEOUT;
+    }
 
-        return serverTimeout;
+    public TimeBudget readBudget() {
+        // timeLeft = CONNECT_TIMEOUT + readTimeout
+        return timeBudget.withReserved(CONNECT_TIMEOUT);
```

Comment:
Throw if non-positive?

Reply:
It does that when you read from the budget.